core/validation/specific_validations/towns_fund_round_four.py

```diff
@@ -500,6 +501,70 @@ def validate_postcodes(workbook: dict[str, pd.DataFrame]) -> list["TownsFundRoun
         ]
 
 
+def validate_current_project_delivery_status_and_upcoming_comms_milestones(
+    workbook: dict[str, pd.DataFrame]
+) -> list["TownsFundRoundFourValidationFailure"] | None:
+    """
+    Validates the Project Progress table in the Round 4 submission workbook.
+
+    Checks for rows with a Project Delivery Status other than "4. Completed" where both
+    the Current Project Delivery Stage and Most Important Upcoming Comms Milestone are null.
+    If such invalid states are found, returns a list of TownsFundRoundFourValidationFailure instances.
+
+    :param workbook: A dictionary containing sheet names as keys and pandas DataFrames
+                     representing each sheet in the Round 4 submission.
+    :return: A list of TownsFundRoundFourValidationFailure instances if validation failures occur,
+             otherwise returns None.
+    """
+    project_progress_df = workbook["Project Progress"]
+
+    project_incomplete_rows = project_progress_df[project_progress_df["Project Delivery Status"] != "4. Completed"]
+
+    na_values = {"", "< Select >", np.nan, None}
```

Comment:
I think it can, I suppose the question is whether we want to have the ```na_values``` adjustable or not. For example, for dropdowns we want "< Select >" to be null, for non-dropdowns we don't both with the value. There might be a more useful use case that arises in future development, not sure.

Are there instances where np.nan will not catch a null value that is only picked up by pd.NA? In our extraction scripts we default null values to np.nan. Have you discovered null values appearing as pd.NA? If so we need to change the code for all null validation.

Reply:
Correct me if I'm wrong but `< Select >` is removed at transformation time I think?

Reply:
I agree with Connor here - I think we want consistency. Do we create a set of NA values like this anywhere else? Can we define once and reference instead if so